### ISSUE TITLE
uuid collisions

### ISSUE THREAD

**Issue by Jancorhw** [thenativeweb/uuidv4#1]
Hi,
For some reason I notice that the uuids that are generated will repeat after a while. Is this natural to uuidv4 or is this an implementation problem?
The same results happen when run on nodejs v9.2.0 and v10.1.0. I have not tested this with other nodejs versions.

// Script
const uuid = require('uuidv4')

for (let i=0; i < 100000000000; i++) {
    let str = uuid()
    if (/^cacaca/g.test(str)) {
        console.log(new Date(), str)
    }
}

// Output
2018-06-11T14:01:17.878Z 'cacacab3-022d-48c8-9d9d-f05c42b5862a'
2018-06-11T14:02:35.680Z 'cacaca76-f041-4fcb-8ea1-3976147376c3'
2018-06-11T14:02:37.495Z 'cacaca07-b648-4ef2-9159-afa88c6a80f2'
2018-06-11T14:03:01.127Z 'cacacab3-022d-48c8-9d9d-f05c42b5862a'

**Comment by goloroden** [comment]
Hi @Jancorhw 😊 

Thanks a lot for pointing out this issue, this definitely shouldn't happen!
We are going to solve this issue in the short run.

**Comment by goloroden** [comment]
This has been fixed in `1.0.1`, which is now live on npm.

**Comment by goloroden** [comment]
Hi @Jancorhw 😊 

The underlying problem was that we used `Math.random()` to create the random numbers used for the UUIDs, but those numbers are only pseudo-random. I.e., they are not random from a cryptographic point of view, and hence should not be used for something such as creating a UUID.

We have now replaced the algorithm (which was one from StackOverflow) with one that actually uses strong cryptographic random numbers, if available on the appropriate platform. Internally, we now use the [uuid](https://github.com/kelektiv/node-uuid) module for this. This allows us to use the better algorithm, but keep our API.

The update has been released as `1.0.1` to npm, and the problem you described can not be reproduced with this version any longer.

Once again, thanks for pointing this out 👍
